Updated documentation for `Toast` component (#2261)
iTwin/iTwinUI: apps/website/src/content/docs/toast.mdx

---
title: Toast
description: A toast is a non-disruptive message that provides feedback based on a user action.
thumbnail: Toast
---

<p>{frontmatter.description}</p>

## Usage

> Before triggering a toast, the toaster needs to be initialized using the `useToaster` hook. This hook returns an object which provides various methods for triggering the toast and changing the toaster settings.

Reviewer: Simplification + accuracy. (I realized that "initialized" is incorrect, because `useToaster` is simply grabbing a reference to the already-initialized toaster)

```suggestion
Before triggering a toast, you can need to call the `useToaster` hook, which returns an object which provides various methods for triggering the toast and changing the toaster settings.
```

<LiveExample src='Toast.main.jsx'>
  <AllExamples.ToastMainExample client:load />
</LiveExample>

> 

Reviewer: Would be good to mention this caveat, since it is a common issue that our users run into.

```suggestion

**Note**: `useToaster` must be called from the top level of a component which is rendered a descendant of [`ThemeProvider`](/docs/themeprovider).

```

### Status

> Toast statuses can be set using the status methods provided by the toaster object. The status method takes in a toast message and an `options` object for further customization. Four available statuses are:

Reviewer: nit: We are not setting a toast status, we are triggering a toast.

```suggestion
Toasts can be triggered using the status methods provided by the toaster object. The status method takes in a toast message and an `options` object for further customization. Four available statuses are:
```

- `positive()`
- `negative()`
- `warning()`
- `informational()`

<LiveExample src='Toast.status.jsx'>
  <AllExamples.ToastStatusExample client:load />
</LiveExample>

For further customization with `options`, you can utilize `link`, which allows taking further actions directly from the toast, such as navigating to another page or triggering a specific function.

<LiveExample src='Toast.link.jsx'>
  <AllExamples.ToastLinkExample client:load />
</LiveExample>

### Settings

The object returned by `useToaster` also provides a `setSettings` method, which can be used to customize the `placement` and `order` of toasts.

`placement` supports the following values:

- `"top"` (default)
- `"top-start"`
- `"top-end"`
- `"bottom"`
- `"bottom-start"`
- `"bottom-end"`

`"start"` indicates the left side of viewport, while `"end"` represents the right side.

<LiveExample src='Toast.placement.jsx'>
  <AllExamples.ToastPlacementExample client:load />
</LiveExample>

`order` supports the following values:

- `"auto"` (default)
- `"descending"`
- `"ascending"`

> When set to `"descending"`, the most recent toasts are on top. When set to `"ascending"`, the most recent toasts are on bottom. When set to `"auto"`, it will behave like `"descending"` when `placement` is set to a `-top` value, otherwise `"ascending"`.

Reviewer: It represents the "suffix" part of the values, since there is no value equal to `top` exactly.

> When set to `"descending"`, the most recent toasts are on top. When set to `"ascending"`, the most recent toasts are on bottom. When set to `"auto"`, it will behave like `"descending"` when `placement` is set to a `-top` value, otherwise `"ascending"`.

Reviewer: It was supposed to represent the "suffix" part of the values, since `top` is not the only value. It seems like I may have gotten things confused though. It should be a prefix, not a suffix.

Even better suggestion, we could rephrase this as:

```suggestion
When set to `"descending"`, the most recent toasts are on top. When set to `"ascending"`, the most recent toasts are on bottom. When set to `"auto"`, it will behave like `"descending"` when `placement` is set to a value beginning with `"top"`, otherwise it will behave like `"ascending"`.
```

> When set to `"descending"`, the most recent toasts are on top. When set to `"ascending"`, the most recent toasts are on bottom. When set to `"auto"`, it will behave like `"descending"` when `placement` is set to a `-top` value, otherwise `"ascending"`.

Author: I have rephrased this as suggested.

<LiveExample src='Toast.order.jsx'>
  <AllExamples.ToastOrderExample client:load />
</LiveExample>

### Closing toasts

> When setting the `hasCloseButton` option available in the status method to `true`, each toast will be displayed with a close button, which allows you to close the toast manually.

Reviewer: Prefer "end user", because "you" typically refers to the developer who is using our components.

```suggestion
When setting the `hasCloseButton` option available in the status method to `true`, each toast will be displayed with a close button, which allows the end user to close the toast manually.
```

Author: I have all the wording/text changes.

<LiveExample src='Toast.hasCloseButton.jsx'>
  <AllExamples.ToastHasCloseButtonExample client:load />
</LiveExample>

> Additionally, to implement further actions upon each toast closing or chain after one another to display multiple messages, you can retrieve the `close()` function returned from the status method.

Reviewer: I think there's a typo in your suggestion.

> Additionally, to implement further actions upon each toast closing or chain after one another to display multiple messages, you can retrieve the `close()` function returned from the status method.

Author: I have made the change.

> Additionally, to implement further actions upon each toast closing or chain after one another to display multiple messages, you can retrieve the `close()` function returned from the status method.

Author: I have made the change.

> Additionally, to implement further actions upon the closing of each toast or to chain one toast after another, you can retrieve the `close()` function returned from the status method.

Reviewer: ```suggestion
Additionally, to implement further actions upon the closing of each toast or to chain one toast after another, you can call the `close()` function returned from the status method.
```

<LiveExample src='Toast.close.jsx'>
  <AllExamples.ToastCloseExample client:load />
</LiveExample>

> Closing also depends on which type of toast you are using, without the use of `hasCloseButton`. Persisting toasts will not be closed automatically, and will contain a close button. Temporary toasts will automatically close after 7 seconds and will not contain a close button. The type of toasts can be specified by passing either `"persisting"` or `"temporary"` into the `type` option.

Author: I have done as suggested.

<LiveExample src='Toast.type.jsx'>
  <AllExamples.ToastTypeExample client:load />
</LiveExample>

> To close all toasters on viewport at once, use the `closeAll()` method offered by the toaster object.

Reviewer: ```suggestion
To close all currently open toasts at once, use the `closeAll()` method offered by the toaster object.
```

<LiveExample src='Toast.closeAll.jsx'>
  <AllExamples.ToastCloseAllExample client:load />
</LiveExample>

> The `animateOutTo` prop allows you to control the direction to which the toast animates out when closing. Specifically, it helps determine an "anchor" element that the toast will go towards when it is dismissed by pointing to a ref passed into that element.

Reviewer: We can simplify wording and provide an example use-case of this feature.

```suggestion
The `animateOutTo` prop allows you to control the direction to which the toast animates out when closing. Specifically, it helps determine an "anchor" element that the toast will go towards when it is dismissed. This can be useful, for example, when you want to indicate to the user that the toasts are being saved into a "notifications" panel.
```

---
title: Toast
description: A toast is a non-disruptive message that provides feedback based on a user action.
thumbnail: Toast
---

<p>{frontmatter.description}</p>

## Usage

Before triggering a toast, you need to call the `useToaster` hook, which returns an object containing various methods for triggering the toast and changing the toaster settings.

<LiveExample src='Toast.main.jsx'>
  <AllExamples.ToastMainExample client:load />
</LiveExample>

> **Note**: `useToaster` must be called from the top level of a component which is rendered a descendant of [`ThemeProvider`](/docs/themeprovider).

Author: I have made the change :)

---

<p>{frontmatter.description}</p>

## Usage

Initialize your toast using the `useToaster` hook, and customize its `placement`, `order` and other options via the `setSettings` method. To specify the content of the toast, pass the toast message into the status method along with further control options.

<LiveExample src='Toast.main.jsx'>
  <AllExamples.ToastMainExample client:load />
</LiveExample>

### Status

There are four available statuses that comes with toast message:

- `positive`
- `negative`
- `warning`
- `informational`

<LiveExample src='Toast.status.jsx'>
  <AllExamples.ToastStatusExample client:load />
</LiveExample>

### Link

The toast notification can come with a hyperlink using the `link` prop. You can utilize this prop to take further action directly from the toast, such as navigating to another page or triggering a specific function.

<LiveExample src='Toast.link.jsx'>
  <AllExamples.ToastLinkExample client:load />
</LiveExample>

### Positioning

The toast position can be customized using the `placement` prop, which is default to `top`. The available `placement` values are:

- `top`
- `top-start`
- `top-end`
- `bottom`
- `bottom-start`
- `bottom-end`

> `start` indicates left side of viewport, while `end` represents the right side of viewport.

Author: I have made the change.

<p>{frontmatter.description}</p>

<LiveExample src='Toast.main.jsx'>
  <AllExamples.ToastMainExample client:load />
</LiveExample>

A toast notification provides a brief message about app processes at the top of the screen. Because these messages are so prominent, we need to be careful not to overuse or misuse them.

> ## Variants

Author: I have removed the "Variants" section and moved everything under "Usage".

> 

Author: I have had a section for `link`.

### Status

A toast message can come with a status such as:

- `positive`
- `negative`
- `warning`
- `informational`

<LiveExample src='Toast.positive.jsx'>
  <AllExamples.ToastPositiveExample client:load />
</LiveExample>

<LiveExample src='Toast.negative.jsx'>
  <AllExamples.ToastNegativeExample client:load />
</LiveExample>

<LiveExample src='Toast.warning.jsx'>
  <AllExamples.ToastWarningExample client:load />
</LiveExample>

> <LiveExample src='Toast.informational.jsx'>

Author: I have combined them all in one `Live Example`.

  <AllExamples.ToastInformationalExample client:load />
</LiveExample>

### Position changed

The toast position can be customized using the `placement` prop, which is default to `top`. The available `placement` values are:

> - `bottom-end`

Author: I have made this change.

- `top`
- `top-start`
- `top-end`
- `bottom`
- `bottom-start`

> `Start` indicates left side of viewport. `End` - right side of viewport.

Author: I have made the change.

<LiveExample src='Toast.positionChanged.jsx'>
  <AllExamples.ToastPositionChangedExample client:load />
</LiveExample>

### Anchor to button

> Use `animateOutTo` to set the direction to which the toast goes when closing.

Author: I have described more about `animateOutTo`.

<LiveExample src='Toast.anchorToButton.jsx'>
  <AllExamples.ToastAnchorToButtonExample client:load />
</LiveExample>

## Usage

> ### Close individual

Author: I have combined those that are related to closing all under "Closing toasts".

> Each toast can be closed individually and chained after one another to display multiple messages.

Author: I have explained more about closing toasts.
